src/util/pelz_io.c

```diff
@@ -84,34 +83,38 @@ int get_file_ext(charbuf buf, int *ext)
     {
       if (memcmp(buf.chars + period_index, ext_type[i], strlen(ext_type[i])) == 0)
       {
-        *ext = i + 1;
-        break;
+        return (i + 1);
       }
     }
   }
-  return 0;
+  return NO_EXT;
 }
 
-int key_load(size_t key_id_len, unsigned char *key_id, size_t * key_len, unsigned char **key)
+int key_load(charbuf key_id)
 {
+  charbuf key;
   int return_value = 1;
+  TableResponseStatus status;
   UriUriA key_id_data;
+  int ext = NO_EXT;
 
   const char *error_pos = NULL;
   char *key_uri_to_parse = NULL;
 
+  ext = get_file_ext(key_id);
```

Comment:
Should we check that we found an extension here and error out if not?

Reply:
The ```git_file_ext``` function may need to be renamed because it is not actually getting the file ext anymore.  It is checking if the file has a ```.nkl``` or a ```.ski``` ext  otherwise it returns a ```NO_EXT```. The ```NO_EXT``` just means that we don't treat the file like a ```.nkl``` or a ```.ski``` whether it has an ext or not. The ```return``` is an ```ExtentionType``` whether it fails or not.